### PR TITLE
[IMP] test_lint: detect useless manifest content

### PR DESCRIPTION
**Pull request by Feyensv**
Keep the manifests as light as possible, to easily see custom behavior/content.
Complete the work of previous commits cleaning the manifests content:

* https://github.com/odoo/odoo/commit/42bad1a6d21e086e3bb766f9813d65902b3c232a
* https://github.com/odoo/odoo/commit/ef7005f52416912a9efab5ec2767432c4181fb4d

and make sure this kind of cleanup commit is not necessary in the future
because it is now automatically verified by a dedicated test.

Enterprise PR: https://github.com/odoo/enterprise/pull/34903

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/test_lint/tests/test_manifests.py [hunk 1]
```diff
@@ -13,9 +16,46 @@
...
+        mod_path = get_module_path(module, downloaded=True)
+        manifest_file = module_manifest(mod_path)
+
+        manifest_data = {}
+        with file_open(manifest_file, mode='r') as f:
+            manifest_data.update(literal_eval(f.read()))
+
+        return manifest_data
+
     def test_manifests_keys(self):
```
Comment (Julien00859) [inline odoo/addons/test_lint/tests/test_manifests.py:34]:
```suggestion
    def test_manifests(self):
```

**Review by Julien00859 — COMMENTED** [review]
Hey Victor, since we are at it, can you also add a test case that verify mandatory entries are set and non-empty ?

cc https://github.com/odoo/odoo/pull/107216#issuecomment-1348504087

**Comment by Feyensv** [comment]
> Hey Victor, since we are at it, can you also add a test case that verify mandatory entries are set and non-empty ?
> 
> cc [#107216 (comment)](https://github.com/odoo/odoo/pull/107216#issuecomment-1348504087)

There you go, some more checks (typing and making sure icon is not set or set with a truthy value).
Did you have any other 'mandatory' key in mind ?

**Comment by Julien00859** [comment]
Those listed at the top:
```
MANIFEST_KEYS = {
    'name', 'icon', 'addons_path', 'license',  # mandatory keys
```

**Comment by Feyensv** [comment]
> Those listed at the top:

`name` is already required on `ir.module.module` model, and `license` is already enforced here:
https://github.com/odoo/odoo/blob/dfd5eea60b71c5759442936b32a8d8d33093b1ce/odoo/modules/module.py#L317-L319

and for addons-path logic, considering we rely on standard behavior in our modules and never specify it, i don't think it's really useful to check it here :thinking:

**Review by Julien00859 — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**